src/ui/tabbar.ts

```diff
@@ -186,6 +186,17 @@ class TabBar extends Widget {
   tabMoved: ISignal<TabBar, TabBar.ITabMovedArgs>;
 
   /**
+   * A signal emitted when a tab is clicked by the user.
+   *
+   * #### Notes
+   * If the clicked tab is not the current tab, the clicked tab will be
+   * made current and the `currentChanged` signal will be emitted first.
+   *
+   * This signal is emitted even if the clicked tab is the current tab.
+   */
+  tabActivateRequested: ISignal<TabBar, TabBar.ITabActivateRequestedArgs>;
```

Comment:
I'm trying to avoid emitting events as signals. The signal should have semantic meaning, rather than eventful meaning.